Client/Application/NavigationRouter.swift

```diff
@@ -140,12 +140,20 @@ enum NavigationPath {
 
     static func handle(nav: NavigationPath, with bvc: BrowserViewController) {
         switch nav {
-        case .fxa(let params): NavigationPath.handleFxA(params: params, with: bvc)
-        case .deepLink(let link): NavigationPath.handleDeepLink(link, with: bvc)
-        case .url(let url, let isPrivate): NavigationPath.handleURL(url: url, isPrivate: isPrivate, with: bvc)
-        case .text(let text): NavigationPath.handleText(text: text, with: bvc)
-        case .glean(let url): NavigationPath.handleGlean(url: url)
-        case .closePrivateTabs: NavigationPath.handleClosePrivateTabs(with: bvc)
+        case .fxa(let params):
+            NavigationPath.handleFxA(params: params, with: bvc)
+        case .deepLink(let link):
+            NavigationPath.handleDeepLink(link, with: bvc)
+        case .url(let url, _):
+            // Note: The we get browser private mode from current selected tab
+            let isPrivate = bvc.tabManager.selectedTab?.isPrivate ?? false
+            NavigationPath.handleURL(url: url, isPrivate: isPrivate, with: bvc)
```

Comment:
This is the only change required however I modified other areas to match the style

Reply:
Should we update the userDefaults key for `wasLastSessionPrivate` here, or remove it altogether from the url case?

Reply:
No, I was actually thinking of using this but update the proper state. Going to push another commit if my idea works 🤞🏼